From 780aa6681f7f87e322d51959c22e3998c75b1f42 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Tue, 17 Jun 2014 21:17:21 +0800
Subject: [PATCH 2/4] devicetree.py: don't check rootfs' partition for
 upgrading

We should not check the partition which the rootfs is running on for
upgrading. It won't check if the rootfs is running on Live OS or USB,
but it would check if it running on a disk image such as the vmdk, this
patch disables it.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/storage/devicetree.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/pyanaconda/storage/devicetree.py b/pyanaconda/storage/devicetree.py
index cce8481..57402cd 100644
--- a/pyanaconda/storage/devicetree.py
+++ b/pyanaconda/storage/devicetree.py
@@ -811,6 +811,14 @@ class DeviceTree(object):
             # faulty partitions.
             return
 
+        # Protect the rootfs' partition
+        root_st_dev = os.stat('/').st_dev
+        major = udev_device_get_major(info)
+        minor = udev_device_get_minor(info)
+        root_major = os.major(root_st_dev)
+        root_minor = os.minor(root_st_dev)
+        if major == root_major and minor == root_minor:
+            device.protected = True
         self._addDevice(device)
         return device
 
-- 
1.8.3.1

